components/ntp_background_images/browser/ntp_sponsored_images_data.cc

```diff
@@ -145,25 +148,70 @@ NTPSponsoredImagesData::NTPSponsoredImagesData(
     return;
   }
 
+  url_prefix = base::StringPrintf("%s://%s/", content::kChromeUIScheme,
+                                  kBrandedWallpaperHost);
   if (auto* name = json_value->FindStringKey(kThemeNameKey)) {
     theme_name = *name;
-    DVLOG(2) << __func__ << ": Theme name: " << theme_name;
+    url_prefix += kSuperReferralPath;
+  } else {
+    url_prefix += kSponsoredImagesPath;
+  }
+
+  auto* campaigns_value = json_value->FindListKey(kCampaignsKey);
+  if (campaigns_value) {
+    ParseCampaignsList(campaigns_value->Clone(), installed_dir);
```

Comment:
Can't we pass as reference without having to clone the data?

Reply:
Yup, can do. Done by https://github.com/brave/brave-core/pull/11628